### ISSUE TITLE
Bugs of ol4j.mdx.parser  

### ISSUE THREAD

**Issue by chenlini** [olap4j/olap4j#55]
Hello, i'm doing a project about MDX parser.
When I use MdxParser to parse a MDX statement to a parseTree, I find some bugs. 
Those statements contain follow parts can not be parsed.
1. "Having"
2. "("xxx")"
3. "$"

**Comment by julianhyde** [comment]
You are correct that the parser doesn't handle the HAVING clause. We would welcome a contribution that could handle that.

The other two examples don't look like valid MDX to me.
